atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/snapshot/DefaultKeyValueSnapshotReaderManager.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl.snapshot;
+
+import com.palantir.atlasdb.cell.api.TransactionKeyValueService;
+import com.palantir.atlasdb.cell.api.TransactionKeyValueServiceManager;
+import com.palantir.atlasdb.transaction.api.DeleteExecutor;
+import com.palantir.atlasdb.transaction.api.OrphanedSentinelDeleter;
+import com.palantir.atlasdb.transaction.api.snapshot.KeyValueSnapshotReader;
+import com.palantir.atlasdb.transaction.api.snapshot.KeyValueSnapshotReaderManager;
+import com.palantir.atlasdb.transaction.impl.ReadSentinelHandler;
+import com.palantir.atlasdb.transaction.service.TransactionService;
+
+public class DefaultKeyValueSnapshotReaderManager implements KeyValueSnapshotReaderManager {
```

Comment:
nit: The `Default*` classes usually are `final`, we don't really want people to extend this

Reply:
This should be flagged by some tool

Reply:
https://github.com/palantir/atlasdb/pull/7145

Reply:
There's an errorProne check that should enforce this, but it's disabled on the repo because we have a lot of violations.

Reply:
Yeah, `DesignForExtension` I think it's called. Unfortunately there are too many historic violations (atlas predates these checks)